Mathlib/Algebra/Homology/DerivedCategory/Ext/Basic.lean

```diff
@@ -422,6 +421,68 @@ noncomputable def extFunctor (n : ℕ) : Cᵒᵖ ⥤ C ⥤ AddCommGrp.{w} where
     apply Ext.comp_assoc
     all_goals omega
 
+section biproduct
+
+attribute [local simp] Ext.mk₀_add
+
+instance (X : C) (n : ℕ) : (extFunctorObj X n).Additive where
+
+instance (n : ℕ) : (extFunctor (C := C) n).Additive where
+
+lemma Ext.comp_sum {X Y Z : C} {p : ℕ} (α : Ext X Y p) {ι : Type*} [Fintype ι] {q : ℕ}
+    (β : ι → Ext Y Z q) {n : ℕ} (h : p + q = n) :
+    α.comp (∑ i, β i) h = ∑ i, α.comp (β i) h :=
+  map_sum (α.precomp Z h) _ _
+
+lemma Ext.sum_comp {X Y Z : C} {p : ℕ} {ι : Type*} [Fintype ι] (α : ι → Ext X Y p) {q : ℕ}
+    (β : Ext Y Z q) {n : ℕ} (h : p + q = n) :
+    (∑ i, α i).comp β h = ∑ i, (α i).comp β h :=
+  map_sum (β.postcomp X h) _ _
+
+lemma Ext.mk₀_sum {X Y : C} {ι : Type*} [Fintype ι] (f : ι → (X ⟶ Y)) :
+    mk₀ (∑ i, f i) = ∑ i, mk₀ (f i) :=
+  map_sum addEquiv₀.symm _ _
+
+variable {J : Type*} [Fintype J] {X : J → C} {c : Bicone X} (hc : c.IsBilimit) (Y : C) (n : ℕ)
+
+/-- `Ext` commutes with biproducts in its first variable. -/
+noncomputable def Ext.biproductAddEquiv : Ext c.pt Y n ≃+ Π i, Ext (X i) Y n where
+  toFun e i := (Ext.mk₀ (c.ι i)).comp e (zero_add n)
+  invFun e := ∑ (i : J), (Ext.mk₀ (c.π i)).comp (e i) (zero_add n)
+  left_inv x := by
+    simp only [← comp_assoc_of_second_deg_zero, mk₀_comp_mk₀]
+    rw [← Ext.sum_comp, ← Ext.mk₀_sum, IsBilimit.total hc, mk₀_id_comp]
+  right_inv _ := by
+    ext i
+    simp only [Ext.comp_sum, ← comp_assoc_of_second_deg_zero, mk₀_comp_mk₀]
+    rw [Finset.sum_eq_single i _ (by simp), bicone_ι_π_self, mk₀_id_comp]
+    intro _ _ hij
+    rw [c.ι_π, dif_neg hij.symm, mk₀_zero, zero_comp]
+  map_add' _ _ := by
+    simp only [comp_add]
+    rfl
+
+variable (X : C) {J : Type*} [Fintype J] {Y : J → C} {c : Bicone Y} (hc : c.IsBilimit) (n : ℕ)
```

Comment:
Some variables are introduced multiple times. In general, we can use sections, here you may declare some of the variables directly in the definitions.